samples/error-correction/syndrome/Syndrome.qs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.Samples.Hardware.Syndrome
+{
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Preparation;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Core;
+
+    /// # Summary
+    /// Apply a controlled Pauli operation for a given basis and control/target qubits
+    ///
+    /// # Input
+    /// ## basis
+    /// Pauli basis in which to apply the controlled pauli
+    /// ## control
+    /// Control qubit
+    /// ## target
+    /// Target qubit
+    operation ControlledPauli(basis: Pauli, control: Qubit, target: Qubit): Unit {
```

Comment:
Could this be implemented as `Controlled ApplyPauli(control, ([basis], [target]));` instead of using the three `if` statements?

Reply:
That's neat, thanks for the suggestion!